### PR TITLE
Changes made for Learnathon

### PR DESCRIPTION
**Pull request by kozusznik**
Changes as ParallelTask renaming should be helpfull

### CODE REVIEW

File: src/main/java/cz/it4i/parallel/ImageJServerWorker.java [hunk 1]
```diff
@@ -46,9 +46,14 @@
 	private final static Set<String> supportedImageTypes = Collections
 			.unmodifiableSet(new HashSet<>(Arrays.asList("png", "jpg")));
 
-	ImageJServerWorker(String hostName, int port) {
+	ImageJServerWorker(String hostName, int defaultPort) {
 		this.hostName = hostName;
-		this.port = port;
+		if(hostName.contains(":")) {
```
Comment (PetrBainar) [inline src/main/java/cz/it4i/parallel/ImageJServerWorker.java]: I think that this logic should either be moved upstream to initWorkerPool().

File: src/test/java/test/learnathon/Demonstration.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		hosts.add("localhost:10003");
+		hosts.add("localhost:10004");
+		((ImageJServerParadigm) paradigm).setHosts(hosts);
+		
+		// common initialization
+		paradigm.init();
+
+		paradigm.parallelFor(angles, (angle, executionContext) -> {
+			log.info("processing angle=" + angle);
+			IntStream.iterate(0, n->n+step).takeWhile(n->n<10);
```
Comment (PetrBainar) [inline src/test/java/test/learnathon/Demonstration.java]: This is a Java 1.9 method. As we are targeting the 1.8 environment, I would prefer to refrain from using this method.

**Review by PetrBainar — CHANGES_REQUESTED** [review]

File: src/test/java/test/learnathon/Demonstration.java (hunk 2 above)
Comment (kozusznik) [reply src/test/java/test/learnathon/Demonstration.java]: Yes, I've  forgotten to remove it.

File: src/main/java/cz/it4i/parallel/ImageJServerWorker.java (hunk 1 above)
Comment (kozusznik) [reply src/main/java/cz/it4i/parallel/ImageJServerWorker.java]: You, are right - it was easier to implement it in the worker.

**Review by PetrBainar — APPROVED** [review]

**Pull request merged**